drivers/flash/flash_stm32_qspi.c

```diff
@@ -323,7 +344,13 @@ static int flash_stm32_qspi_read(const struct device *dev, off_t addr,
 	};
 
 	qspi_set_address_size(dev, &cmd);
-	qspi_prepare_quad_read(dev, &cmd);
+	if (IS_ENABLED(STM32_QSPI_USE_QUAD_IO)) {
+		ret = qspi_prepare_quad_read(dev, &cmd);
```

Comment:
the test of  the STM32_QSPI_USE_QUAD_IO is done out of the function
and then the `qspi_prepare_quad_read() `function itself checks the `dev_data->flag_quad_io_en`
compare to previoulsy "if (IS_ENABLED(STM32_QSPI_USE_QUAD_IO) && dev_data->flag_quad_io_en) "
Correct ?

Reply:
Yes, the `IS_ENABLED` is moved outside.

What do you think about removing `flag_quad_io_en`? I don't think there is a need for it. 
If the driver is in Quad mode (`STM32_QSPI_USE_QUAD_IO` is enabled), then the `init` will fail if QE enabling fails... So there won't be a case that QE enabling failed and we still use the QSPI in Quad mode...

Reply:
I agree